### PR TITLE
Support "all lines attached to stem" beam style of tremolo (partially fix #18897)

### PR DESCRIPTION
**Pull request by Howard-C**
Partially resolves: https://musescore.org/en/node/18897.

This enables the "all lines attached to stem" beam style for (only) minim two-note tremolo. It's one of the most common things used in music notation which MuseScore currently doesn't support. The style can be changed in the inspector, and can also become a score style.

image
image

There should be another beam style for minim two-note tremolo, with only one (top/bottom, depending on stem direction) line attached to stem, but it's much more complicated than the former case, and less common too (I haven't seen that once but seen plenty of the "all lines attached" style). This doesn't mean it shouldn't be supported, but the "all lines attached" style option is much more requested and should probably be supported ASAP.

### CODE REVIEW

**Comment by Howard-C** [comment]
It seems I can only request review from a limited number of people on a (random?) short list... Sorry @Jojo-Schmitz and others ;-)

**Comment by Jojo-Schmitz** [comment]
Only from those with commit rights, I assume?

**Comment by Howard-C** [comment]
> Only from those with commit rights, I assume?

Wow! Didn't realize there're a lot more people with commit rights!

**Comment by Jojo-Schmitz** [comment]
> > Only from those with commit rights, I assume?
> 
> Wow! Didn't realize there're a lot more people with commit rights!

Well, that list is historically grown, some never or rarely made use of that right, some won't anymore going forward. And I see 9, than's not that many

**Comment by MarcSabatella** [comment]
This is super cool, thanks for implementing it!  I haven't tested, but it looks liek things are basically in place that would need to be in terms of managing the property/style, and I'm pleased that the actual layout changes are so small and self-contained.

One thing I wonder: would it make sense to also look at https://musescore.org/en/node/20390, which will probably touch on some of the same code?  Not saying that couldn't be dealt with entirely separately, but I'd guess there would be a high likelihood of merge conflicts if these were two separate PR's.

**Comment by Howard-C** [comment]
I will look into that soon, thanks! 
As you can see, that issue deals with single-note tremolos while this PR deals with two-note tremolo. So there shouldn't be many merge conflicts, thus it's better to create a separate PR for that.

**Comment by Howard-C** [comment]
See #5619.

File: libmscore/tremolo.h [hunk 1]
```diff
@@ -44,7 +48,8 @@ class Tremolo final : public Element {
       QPainterPath path;
 
       int _lines;       // derived from _subtype
-      TremoloPlacement _tremoloPlacement = TremoloPlacement::DEFAULT;
+      TremoloPlacement _tremoloPlacement;
+      TremoloBeamStyle _beamStyle;
```
Comment (igorkorsukov) [inline libmscore/tremolo.h:52]: Better to always initialize. Why is the initialization removed?

Comment (Howard-C) [reply libmscore/tremolo.h:52]: Don't know... I'll add it back and also initialize `_beamStyle`.

Comment (Howard-C) [reply libmscore/tremolo.h:52]: Oh I see, it's probably because all other properties are uninitialized. I can add initialization to all of them.

**Comment by Jojo-Schmitz** [comment]
Question from the Translarors' chat: "all lines attached to stem", isn't it rather "...stems" i.e. plural, as there are 2 stems involved? Also the rest of the inspector talks about (Tremolo) Beam, so shouldn't that "...lines..." rather be "...beams...?

**Comment by Howard-C** [comment]
> Question from the Translarors' chat: "all lines attached to stem", isn't it rather "...stems" i.e. plural, as there are 2 stems involved? 

I find this reasonable too... I will change this.

> so shouldn't that "...lines..." rather be "...beams...?

The word "beam" seems more like a summary of the lines, so I won't use the plural form "beams" here. Also tremolo strokes is not a "beam", it has "beaming styles" but is not a beam. So probably "strokes" can make sense too, but to me "lines" is a familiar term.

**Comment by Jojo-Schmitz** [comment]
> > Question from the Translarors' chat: "all lines attached to stem", isn't it rather "...stems" i.e. plural, as there are 2 stems involved?
> 
> I find this reasonable too... I will change this.
> 
> > so shouldn't that "...lines..." rather be "...beams...?
> 
> The word "beam" seems more like a summary of the lines, so I won't use the plural form "beams" here. Also tremolo strokes is not a "beam", it has "beaming styles" but is not a beam. So probably "strokes" can make sense too, but to me "lines" is a familiar term.

"strokes" sounds reasonable. But then not only in this one string

**Pull request merged**